proknow/Audit.py

```diff
@@ -0,0 +1,159 @@
+        if "patient_mrn" in kwargs: self.options["patient_mrn"] = kwargs["patient_mrn"]
+        if "workspace_id" in kwargs: self.options["workspace_id"] = kwargs["workspace_id"]
+        if "workspace_name" in kwargs: self.options["workspace_name"] = kwargs["workspace_name"]
+        if "collection_id" in kwargs: self.options["collection_id"] = kwargs["collection_id"]
+        if "resource_id" in kwargs: self.options["resource_id"] = kwargs["resource_id"]
+        if "resource_name" in kwargs: self.options["resource_name"] = kwargs["resource_name"] 
+        if "classification" in kwargs: self.options["classification"] = kwargs["classification"].upper() 
+        if "uri" in kwargs: self.options["uri"] = kwargs["uri"] 
+        if "user_agent" in kwargs: self.options["user_agent"] = kwargs["user_agent"]
+        if "ip_address" in kwargs: self.options["ip_address"] = kwargs["ip_address"]
+
+        if "types" in kwargs: 
+            if not isinstance(kwargs["types"], list):
+                self.options["types"] = [kwargs["types"]]
+            else:
+                self.options["types"] = kwargs["types"]
+
+        if "methods" in kwargs: 
+            if not isinstance(kwargs["methods"], list):
+                self.options["methods"] = [kwargs["methods"].upper()]
+            else:
+                self.options["methods"] = [x.upper() for x in kwargs["methods"]]
+
+        if "status_codes" in kwargs: 
+            if not isinstance(kwargs["status_codes"], list):
+                self.options["status_codes"] = [kwargs["status_codes"]]
+            else:
+                self.options["status_codes"] = kwargs["status_codes"]
+
+        if "text" in kwargs: 
+            if not isinstance(kwargs["text"], list):
+                self.options["text"] = [kwargs["text"]]
+            else:
+                self.options["text"] = kwargs["text"]
+
+        self.auditResultsPage = self._query()
+
+        self.options['page_number'] = 0
+        if self.auditResultsPage.total > 0:
+            self.options['first_id'] = self.auditResultsPage.items[0]['id']
+
+        return self.auditResultsPage
+
+    def next(self):
+        """Gets the next page of results using the parameters provided by the 
+        :meth:`proknow.Audit.Audit.query` method.
+        
+        Returns:
+            :class:`proknow.Audit.AuditResultsPage`
+
+        Raises:
+            :class:`proknow.Exceptions.HttpError`: If the HTTP request generated an error.
+            :class:`proknow.Exceptions.InvalidOperationError`: If this method is called before :meth:`proknow.Audit.Audit.query` is called.
+        """
+        if self.auditResultsPage is None:
+            raise InvalidOperationError('\'Audit.query\' must be called before calling \'Audit.next\'')
+
+        self.options['page_number'] += 1
+        self.auditResultsPage = self._query()
+
+        return self.auditResultsPage
```

Comment:
Once you address my comment above, this method will no longer be needed. You will only need the one for `AuditResultsPage`.